src/App.tsx

```diff
@@ -15,71 +15,76 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state:Readonly<State> = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  delete = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  handler = (name: string) => {
+    this.setState(
+      (prevState) => (
+        { selectedGood: name === prevState.selectedGood ? '' : name }
+      ),
+    );
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render() {
+    return (
+      <main className="section container">
+        {(this.state.selectedGood.length === 0) ? (
+          <h1 className="title">No goods selected</h1>
+        ) : (
+          <>
+            <h1 className="title is-flex is-align-items-center">
+              {this.state.selectedGood}
+              {' is selected'}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                id="button"
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => this.delete()}
```

Comment:
Fix it everywhere please

```suggestion
                onClick={this.delete}
```